src/main/java/org/broadinstitute/hellbender/tools/copynumber/utils/annotatedregion/SimpleAnnotatedGenomicRegion.java

```diff
@@ -59,6 +56,34 @@ public int getEnd() {
         return interval.getEnd();
     }
 
+    public void setInterval(final Locatable interval) {
```

Comment:
Why does this need to be mutable?  Perhaps I will discover why later, but I think most other records (if not all) in the CNV collections are essentially immutable.

Also, wouldn't it make more sense to parameterize as `SimpleAnnotatedGenomicRegion<Annotations>` (or something along those lines)?  Is there anything stopping me from trying to make a `SimpleAnnotatedGenomicRegionCollection` out of a list of `SimpleAnnotatedGenomicRegion`s that have maps with different keys?

I would guess that refactoring along these lines would simplify the code.  Let me read on and see.

Reply:
After looking through the code, I think it needs to be refactored.  The collections framework isn't really used as intended---and there are many places where you actually modify the framework itself to accommodate your new collection.

As stated in the above comment, the use of a mutable map to represent each record in a TSV seems unnecessary and may be problematic if no checks are performed on the keys.  I think making `AnnotatedInterval` more flexible is the better approach.  Although the `AnnotationSet` that is used there currently only contains `GC_CONTENT`, we could perhaps modify it to contain a set of immutable annotations dynamically generated from input columns of interest.

Reply:
I originally did the setters for performance reasons.  But I figured out a way around that.  So I was able to remove setters for the locatable portion.

The discordance of annotation names is still an issue.  I can remove the annotation setter, though.  That would make in-place adjustments impossible, though.  This would hurt performance, since we would be making a lot of copies.  I could also enforce uniform keys in the Collection class, but that might be messy.